### PR TITLE
app-editors/helix: bump to 23.05

### PR DESCRIPTION
**Pull request by freijon**

### CODE REVIEW

**Review by juippis — COMMENTED** [review]
Could you split this into two commits, one for adding a new version and one for removing the old one? I want to review the ebuild since it has few minor things I dislike.

**Comment by freijon** [comment]
Thats's what I did in c896dd5 (add) and 00b57a4 (drop)

File: app-editors/helix/helix-23.05.ebuild [hunk 1]
```diff
@@ -0,0 +1,322 @@
...
+	windows_x86_64_gnullvm-0.42.2
+	windows_x86_64_gnullvm-0.48.0
+	windows_x86_64_msvc-0.42.2
+	windows_x86_64_msvc-0.48.0
+	winnow-0.4.1
+"
+
+inherit bash-completion-r1 cargo desktop xdg-utils
+
+DESCRIPTION="A post-modern text editor."
```
Comment (juippis) [inline app-editors/helix/helix-23.05.ebuild:251]:
```suggestion
DESCRIPTION="A post-modern text editor"
```
https://wiki.gentoo.org/wiki/Project:Proxy_Maintainers/User_Guide/Style_Guide#Descriptive_text

File: app-editors/helix/helix-23.05.ebuild [hunk 2]
```diff
@@ -0,0 +1,322 @@
...
+
+DOCS=(
+	README.md
+	CHANGELOG.md
+	book/
+	docs/
+)
+
+src_compile() {
+	use grammar || local -x HELIX_DISABLE_AUTO_GRAMMAR_BUILD=1
```
Comment (juippis) [inline app-editors/helix/helix-23.05.ebuild:281]: Is there a downside to disabling this flag? Ie should these always be built? It doesn't look like it's pulling any deps.

File: app-editors/helix/helix-23.05.ebuild [hunk 3]
```diff
@@ -0,0 +1,322 @@
...
+
+	cargo_src_compile
+}
+
+src_install() {
+	cargo_src_install --path helix-term
+
+	rm -rf runtime/grammars/.gitkeep || die
+	rm -rf runtime/grammars/sources || die
```
Comment (juippis) [inline app-editors/helix/helix-23.05.ebuild]: Drop the `-f`s. They silently fail leaving possibly redundant calls in the ebuild for ages.

File: app-editors/helix/helix-23.05.ebuild [hunk 4]
```diff
@@ -0,0 +1,322 @@
...
+
+	insinto /usr/share/helix
+	doins -r runtime
+
+	use doc && dodoc -r "${DOCS[@]}"
```
Comment (juippis) [inline app-editors/helix/helix-23.05.ebuild]: Is the size different how huge? Should you always unconditionally install docs? It doesn't look like it's pulling in any deps.

File: app-editors/helix/helix-23.05.ebuild [hunk 5]
```diff
@@ -0,0 +1,322 @@
...
+
+pkg_postinst() {
+	xdg_icon_cache_update
+	xdg_desktop_database_update
+}
+
+pkg_postrm() {
+	xdg_icon_cache_update
+	xdg_desktop_database_update
+}
```
Comment (juippis) [inline app-editors/helix/helix-23.05.ebuild]: You can just inherit *xdg.eclass* and remove these lines for these actions to automatically be called.

**Review by juippis — COMMENTED** [review]
Oops indeed!

File: app-editors/helix/helix-23.03.ebuild [hunk 6]
```diff
@@ -1,334 +0,0 @@
-# Copyright 2023-2023 Gentoo Authors
-# Distributed under the terms of the GNU General Public License v2
-
-# Auto-Generated by helix_ebuild.py (see FILESDIR)
```
Comment (juippis) [inline app-editors/helix/helix-23.03.ebuild:4]: Please also remove this file with this commit.

File: app-editors/helix/helix-23.05.ebuild (hunk 2 above)
Comment (freijon) [reply app-editors/helix/helix-23.05.ebuild:281]: I think most people will want them, but it's an additional 110 MiB of compiled *.so libraries. The user might want to not build all grammars but only the ones he needs at a later time (the app offers this functionality).

File: app-editors/helix/helix-23.05.ebuild (hunk 4 above)
Comment (freijon) [reply app-editors/helix/helix-23.05.ebuild]: +400 KiB. Totally fine for me to remove that flag.

File: app-editors/helix/helix-23.03.ebuild (hunk 6 above)
Comment (thesamesam) [reply app-editors/helix/helix-23.03.ebuild:4]: ... and start to use pycargoebuild or similar.

File: app-editors/helix/helix-23.05.ebuild (hunk 2 above)
Comment (juippis) [reply app-editors/helix/helix-23.05.ebuild:281]: Heh, I see.
```
 * Final size of build directory: 1396884 KiB ( 1.3 GiB)
 * Final size of installed tree:    29776 KiB (29.0 MiB)


[ebuild   R    ] app-editors/helix-23.05::gentoo  USE="grammar* -debug" 0 KiB

 * Final size of build directory: 1509360 KiB (  1.4 GiB)
 * Final size of installed tree:   142216 KiB (138.8 MiB)
```

**Review by juippis — APPROVED** [review]
Thanks a lot!

**Pull request closed without merging**